common/utils.py

```diff
@@ -233,12 +238,17 @@ def hash_to_bytesuri(s):
     return s.encode("ascii").ljust(32 * (len(s) // 32 + 1), b"\0")
 
 
-def publish_file_in_ipfs(file_url, file_dir, ipfs_client):
+def ipfsuri_to_bytesuri(uri):
+    # we should pad string with zeros till closest 32 bytes word because of a bug in processReceipt (in snet_cli.contract.process_receipt)
+    return uri.encode("ascii").ljust(32 * (len(uri) // 32 + 1), b"\0")
+
+
+def publish_file_in_ipfs(file_url, file_dir, ipfs_client, wrap_with_directory=True):
```

Comment:
@pratik-vii please check Will it impact the behavior of any existing call in contract_api

Reply:
wrap_with_directory=True

Reply:
It is not used in the contract API. "wrap_with_directory=True" is needed.